cheevos/cheevos.c

```diff
@@ -1742,31 +1739,44 @@ static void rcheevos_show_game_placard(void)
          number_of_active++;
    }
 
-   /* TODO/FIXME - localize strings */
    if (number_of_core == 0)
-      strlcpy(msg, "This game has no achievements.", sizeof(msg));
+      strlcpy(msg, msg_hash_to_str(MSG_CHEEVOS_GAME_HAS_NO_ACHIEVEMENTS), sizeof(msg));
    else if (!number_of_unsupported)
    {
       if (settings->bools.cheevos_start_active)
          snprintf(msg, sizeof(msg),
-            "All %d achievements activated for this session.",
+            msg_hash_to_str(MSG_CHEEVOS_ALL_ACHIEVEMENTS_ACTIVATED),
             number_of_core);
       else
          snprintf(msg, sizeof(msg),
-            "You have %d of %d achievements unlocked.",
+            msg_hash_to_str(MSG_CHEEVOS_NUMBER_ACHIEVEMENTS_UNLOCKED),
             number_of_core - number_of_active, number_of_core);
    }
    else
    {
+      char number_of_unsupported_msg[64];
+      char main_msg[256];
+      int _len;
+
+      snprintf(number_of_unsupported_msg,
+               sizeof(number_of_unsupported_msg),
+               msg_hash_to_str(MSG_CHEEVOS_UNSUPPORTED_COUNT),
+               number_of_unsupported);
+
       if (settings->bools.cheevos_start_active)
-         snprintf(msg, sizeof(msg),
-            "All %d achievements activated for this session (%d unsupported).",
-            number_of_core, number_of_unsupported);
+         snprintf(main_msg, sizeof(main_msg),
+                  msg_hash_to_str(MSG_CHEEVOS_ALL_ACHIEVEMENTS_ACTIVATED),
+                  number_of_core);
       else
-         snprintf(msg, sizeof(msg),
-            "You have %d of %d achievements unlocked (%d unsupported).",
-            number_of_core - number_of_active - number_of_unsupported,
-            number_of_core, number_of_unsupported);
+         snprintf(main_msg, sizeof(main_msg),
+                  msg_hash_to_str(MSG_CHEEVOS_NUMBER_ACHIEVEMENTS_UNLOCKED),
+                  number_of_core - number_of_active - number_of_unsupported,
+                  number_of_core);
+
+      _len = snprintf(msg, sizeof(msg),
```

Comment:
I'd rather we write directly to msg from the start with carefully placed snprintfs and strlcpys instead of using an intermediary char variable like main_msg instead.

Reply:
If you don't want to print msg back into itself, you could do something like this:
```
      if (settings->bools.cheevos_start_active)
         _len = snprintf(msg, sizeof(msg),
                  msg_hash_to_str(MSG_CHEEVOS_ALL_ACHIEVEMENTS_ACTIVATED),
                  number_of_core);
      else
         _len = snprintf(msg, sizeof(msg),
                  msg_hash_to_str(MSG_CHEEVOS_NUMBER_ACHIEVEMENTS_UNLOCKED),
                  number_of_core - number_of_active - number_of_unsupported,
                  number_of_core);
```

Then using snprintf:
```
       _len += snprintf(msg + _len, sizeof(msg) - _len, " (%s)", number_of_unsupported_msg);
```

Or alternatively, using strlcpy:
```
       _len += strlcpy(msg + _len, " (", sizeof(msg) - _len);
       _len += strlcpy(msg + _len, number_of_unsupported_msg, sizeof(msg) - _len);
       _len += strlcpy(msg + _len, ")", sizeof(msg) - _len);
```

Reply:
Thanks @Jamiras, that's a great idea! I think the first version looks great so I will give that a go - will push that commit now and will squash, hopefully that does the job